src/runtime/local/kernels/CMakeLists.txt

```diff
@@ -21,10 +21,17 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 # The library of pre-compiled CUDA kernels
 if(USE_CUDA AND CMAKE_CUDA_COMPILER)
+
+    set(CUDA_CODEGEN_CPP_FILES)
+    foreach(i RANGE 0 20)
```

Comment:
This looks like a hard coded value that will cause trouble in the future. When does this need adjusting? Maybe document in a comment above this statement.

Reply:
That's simply the amount of files (kernels) that are generated and would need to be incremented for each new kernel. I didn't find any better way to do that as the amount of kernels is not known until the script is run. The script runs after the point where CMake needs to know about those files.

If we don't find any better way this has to be documented and added to the steps to add a new kernel. Maybe you have an idea? My CMake knowledge is admittedly somewhat limited :)

Reply:
We could try to write the number of output files to a text file in the buld directory and [read that in cmake](https://cmake.org/cmake/help/v3.14/command/file.html)?

Reply:
Wouldn't that just move the hard-coded number to somewhere else?

Maybe we should replace the `add_custom_command` call with an `execute_process`, [here](https://cmake.org/cmake/help/latest/command/execute_process.html).

Maybe that already solves the problem cause it runs at configuration time, so before build time.
If not, I think it should be possible to return the list of files created by the script and use `RESULT_VARIABLE` to set the `CODEGEN_CPP_FILES` with that output.

Either way, just speculation and I'll have to try that first.

Reply:
> Wouldn't that just move the hard-coded number to somewhere else?
I didn't think that through :blush:  We need the number for the add_custom_command. So your method with execute_process might be a better idea. Another (less elegant) solution would be to run that custom command once in a "dry run" mode that only determines the number of produced files.

Reply:
> > I didn't think that through 😊  We need the number for the add_custom_command. So your method with execute_process might be a better idea. Another (less elegant) solution would be to run that custom command once in a "dry run" mode that only determines the number of produced files.

Similar idea: 

What do you think about moving the python execution out of `cmake` and into to `build.sh`? That would massively simplify the `CMakeLists.txt`. The files would be generated in the `build/.../kernels/` path before `cmake` is run so we can simply do

```
file(GLOB sources CONFIGURE_DEPENDS "${path}/*.cpp")
```
Would have to double check if this works properly when recompiling single kernels, but that seems like a very simple solution. I'm not sure how many use the `cmake --build` manually, but proper documentation and a note in the `build.sh` should be enough to avoid any problem in such cases.

Reply:
Meaning you don't like that idea? I also run the `cmake` commands manually but wouldn't have a problem with running `python` manually when adding a kernel, but I'm also fine with not splitting that up. I'll try the `execute_process` tomorrow, maybe that solves that problem anyways.

Reply:
In the context of building DAPHNE I'd say the more automation the better. The cmake docs say that ``execut_process`` runs at config time and ``add_custom_command`` at build time. So setting up the file list with the former to consume in the latter should be doable. You could even bake in that cmake for loop in your favorite scripting language. The file list generation is already there in the main function of ``genKernelInst.py``